### PR TITLE
feat: add WooBuyBackSwap

### PR DESCRIPTION
**Pull request by toprince**

### CODE REVIEW

File: contracts/WooBuybackSwap.sol [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    constructor(address _unirouter) {
+        unirouter = _unirouter;
+        // usdc = 0xaf88d065e77c8cC2239327C5EDb3A432268e5831
+        // weth = 0x82aF49447D8a07e3bd95BD0d56f35241523fBab1
+        // woo = 0xcAFcD85D8ca7Ad1e1C6F82F651fA15E33AEfD07b
+        // arb = 0x912CE59144191C1204E64559FE8253a0e49E6548
+
+        // usdc
+        ILBRouter.Path memory usdcPath;
```
Comment (fb-alexcq) [inline contracts/WooBuybackSwap.sol:59]: Better to make this init logic into a private method.

File: contracts/WooBuybackSwap.sol [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+import {BaseAdminOperation} from "./BaseAdminOperation.sol";
+import {ILBRouter} from "./interfaces/ILBRouter.sol";
+import {TransferHelper} from "./util/TransferHelper.sol";
+
+contract WooBuybackSwap is BaseAdminOperation {
+    event SetUniRouterOnBuyBack(address indexed unirouter);
+    using SafeERC20 for IERC20;
+
+    address public unirouter;
```
Comment (fb-alexcq) [inline contracts/WooBuybackSwap.sol:48]: uniRouter -> lbRouter ; same for event name.

File: contracts/WooBuybackSwap.sol [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
```
Comment (fb-alexcq) [inline contracts/WooBuybackSwap.sol:37]: not needed.

File: contracts/WooBuybackSwap.sol [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+import {IERC20, SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
```
Comment (fb-alexcq) [inline contracts/WooBuybackSwap.sol:38]: SafeERC20 not needed.

**Review by fb-alexcq — CHANGES_REQUESTED** [review]
Per inline comments.

**Review by 0xmer1in — APPROVED** [review]

**Review by fb-alexcq — APPROVED** [review]

File: contracts/WooBuybackSwap.sol [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+    mapping(address => address) public oracles;
+
+    uint256 public slippage; // 1 in 10000th: e.g. 100 = 1%; 50 = 0.5%, 10 = 0.1%
+    uint256 public constant SLIP_BASE = 10000;
+    uint256 public oracleTTL = 15 minutes;
+
+    constructor(address _lbrouter) {
+        // arb contract: https://arbiscan.io/address/0xb4315e873dbcf96ffd0acd8ea43f689d8c20fb30#code
+        lbrouter = _lbrouter;
+        slippage = 50;
```
Comment (fb-alexcq) [inline contracts/WooBuybackSwap.sol:68]: 50 --> 300 ?

**Pull request merged**